codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -495,20 +518,22 @@ private void incrementPendingBytes(int numBytes) {
              * Note: this does not flush the {@link ChannelHandlerContext}.
              */
             int write(int allowedBytes) {
-                int before = payload.size();
-                needFlush |= payload.write(Math.max(0, allowedBytes));
-                int writtenBytes = before - payload.size();
                 try {
+                    pendingWriteQueue.remove();
```

Comment:
So it seems like the problem is we have multiple callers of error (flow controller, encoder, promise callbacks) and we are worried about adverse affects from this?  Instead of fixing that here why don't we let the interface call `error` multiple times and just use the `tryFailure` philosophy.  The first error is taken and propagated, and the remaining errors are ignored?  Constantly adding and removing from the queue for this purpose seems a bit strange (and additional overhead).